### ISSUE TITLE
Faulty xml generated when markdown header is too deep.

### ISSUE THREAD

**Issue by ilitzroth** [doxygen/doxygen#11016]
**Describe the bug**
Faulty xml generated when you add markdown header that is too deep:

**test.h**
```
/*!
 * \brief BRIEF
 *
 * ### SECT3
 */
class test{
};
```

Generates this classtest.xml

```
<?xml version='1.0' encoding='UTF-8' standalone='no'?>
<doxygen xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="compound.xsd" version="1.11.0" xml:lang="en-US">
  <compounddef id="classtest" kind="class" language="C++" prot="public">
    <compoundname>test</compoundname>
    <includes refid="bug1_8h" local="no">bug1.h</includes>
    <briefdescription>
      <para>Encapsulates. </para>
    </briefdescription>
    <detaileddescription>
      <para>blah </para>
      <sect3 id="classtest_1autotoc_md0">
        <title>BLAH</title><para>more blah </para>
      </sect3>
    </detaileddescription>
    <location file="bug1.h" line="9" column="1" bodyfile="bug1.h" bodystart="9" bodyend="10"/>
    <listofallmembers>
    </listofallmembers>
  </compounddef>
</doxygen>
```
According to the compound.xsd detaileddescription cannot contain a sect3. It is a descriptionType

```
<xsd:complexType name="descriptionType" mixed="true">
    <xsd:sequence>
      <xsd:element name="title" type="xsd:string" minOccurs="0"/>	    
      <xsd:element name="para" type="docParaType" minOccurs="0" maxOccurs="unbounded" />
      <xsd:element name="internal" type="docInternalType" minOccurs="0" maxOccurs="unbounded"/>
      <xsd:element name="sect1" type="docSect1Type" minOccurs="0" maxOccurs="unbounded" />
    </xsd:sequence>
  </xsd:complexType>
```

**To Reproduce**
Run the provided header file through doxygen with 
```
MARKDOWN_SUPPORT       = YES
GENERATE_XML = YES
```
**Expected behavior**
A valid xml with respect to compound.xml is produced. 

**Version**
1.11.0

**Additional context**
trying to generate texinfo from the xml output of doxygen with xsd: https://www.codesynthesis.com/projects/xsd/
This comes from trying to run thirdparty headers through doxygen.

**Comment by ilitzroth** [comment]
I can also add that setting 
MARKDOWN_SUPPORT = NO
removes the problem.

**Comment by ilitzroth** [comment]
I don't have control over the original headers that contain the "faulty" doxygen comments. But it's not good that doxygen produces xml that is invalid against it's own compound.xsd. 

The additional context should make clear that I am not "writing doxygen comments" but am trying to process doxygen xml to generate texinfo. The xml parser I use is created from the different .xsd's and it seems quite a bit stricter than the python one.

**Comment by doxygen** [comment]
> @albert-github should we change the <sect3> to <section level="3"> analogous to the <heading> although this would be a breaking change (Sphinx / Breathe) or maybe we should change this only when not the full # / ## / ### sequence is present (less breaking as the current version does not follow the XSD).

No I don't think breaking compatibility is a good idea. We should only output valid content, so either add this missing `<sect2>` or ignore the `<sect3>` since it is not inside the correct context. For `<h3>...</h3>` tags we can be less strict as it can be seen as just markup, i.e. similar to `<b>...</b>`.

> @ilitzroth  I don't have control over the original headers that contain the "faulty" doxygen comments. But it's not good that doxygen produces xml that is invalid against it's own compound.xsd.

I do agree with that. Doxygen should provide valid output regardless of the input. Invalid input should either be corrected or ignored.

**Comment by doxygen** [comment]
Decided to go for a fix at the XML output generator, since the other formats seem not to have too much difficulty when some levels are skipped.

@ilitzroth Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by doxygen** [comment]
For reference, the example I used for testing:
```c++
/*!
@mainpage
@tableofcontents

@subsubsection sect3 SECT3

S3 Text

@subsubparagraph sect6_1 SECT6_1

S6_1 Text

@subsubparagraph sect6_2 SECT6_2

S6_2 Text

@paragraph sect4 SECT4

S4 Text

@subsection sect2 SECT2

S2 Text

*/
```

**Comment by albert-github** [comment]
Looks like with this fix we loose some warnings e.g. when having:
```
Just the page

[TOC]

# SECT1_md

### SECT3_md

## SECT2_md

### SECT3_md2
```
we got before the fix the warnings:
```
aa.md:7: warning: Unexpected subsubsection command found inside section!
```
but with the fix there are no warnings at all.

Another (problem with and without the fix) in the HTML TOC:
![image](https://github.com/user-attachments/assets/76be0a29-1351-4033-b939-b42a8e537ae0)

so here `SECT3_md` and `SECT2_md`  are on the same level. In the old situation this is acceptable (due to the warning) but in the new situation this is a bit strange I think.

Example: [example1.tar.gz](https://github.com/user-attachments/files/16348524/example1.tar.gz)

**Comment by doxygen** [comment]
@albert-github The warning become more consistent. Before, a warning would be given if the first invalid section level is an explicit command (e.g. `@subsubsection`) but not when it is auto generated (e.g. `###`). Within a section, the warnings were reversed (i.e. a warning was given for implicitly defined sections with the wrong level, but not for explicitly defined ones). I now only generate warnings for explicitly defined sections (as it was probably intended, but we could also enable the warnings for implicit sections of course).

Example:
```c++
/** @page1 P1
 *  Text
 *
 *  @section s1 Explicit Section
 *  Text
 *
 *  ### Implicit Subsubsection
 *  Text
 */

/** @page2 P2
 *  Text
 *
 *  # Implicit Section
 *  Text
 *
 *  @subsubsection ss2 Explicit Subsubsection
 *  Text
 */

/** @page3 P3
 *  Text
 *
 *  @section s3 Explicit Section
 *  Text
 *
 *  @subsubsection ss3 Explicit Subsubsection
 *  Text
 */

/** @page4 P4
 *  Text
 *
 *  # Implicit Section
 *  Text
 *
 *  ### Implicit Subsubsection
 *  Text
 */

/** @page5 P5
 *  Text
 *
 *  @subsubsection ss Explicit subsubsection
 *  Text
 */

/** @page6 P6
 *  Text
 *  ### Implicit subsubsection
 */
```

**Comment by doxygen** [comment]
> Remains, in my opinion, the TOC problem but this should probably be handled in a different issue.

Indeed. This can now produce invalid HTML output when levels are skipped (e.g. `<ul>` as a direct child of another `<ul>`, so without `<li>`)

**Comment by albert-github** [comment]
> Indeed

1. handled a new issue?
2. fix within this issue?

**Comment by doxygen** [comment]
@albert-github I've added the fix to this issue.

**Comment by albert-github** [comment]
It doesn't look bad
- except for the errors in the tests
- maybe `background-image: none;` should be added in case a level doesn't have a text, will give a bit of empty space but not the down arrow (the empty space will go away with margin-top:0px, in general it looks like that the default top margin in the TOC could be changed from 5px to 0px).

**Comment by doxygen** [comment]
@albert-github regarding the error: I really thought I tested this properly, but seems not. Hopefully should now be ok. 

Regarding the styling: might be an idea indeed, but the current algorithm only sees empty nodes after they are already written, so that needs more work.

**Comment by albert-github** [comment]
Regarding the "arrow", looking in `void HtmlGenerator::writeLocalToc(const SectionRefs &sectionRefs,const LocalToc &localToc)`
it doesn't look like to complex.
Define an extra var with e.g. `N` in case of an empty `si->title();` / `si->label();` and append this to `"<li class=\"level"` or did I miss something?

**Comment by albert-github** [comment]
Did a test with:
- LaTeX: looks OK
- Docbook: 
  ![image](https://github.com/user-attachments/assets/28537400-547f-4927-9138-330b941c8137)
  - looks like TOC is OK
  - paragraph numbering isn't. I think the `1.1.1 SECT3_md` should be numbered `1.1.0.1` (compare with LaTeX)

Note for this test I changed the `[TOC]` to `@tableofcontents{html:6,latex:6,docbook:6}`

**Comment by albert-github** [comment]
I wanted to have a look at the removal of the arrow (and lateer on at the docbook problem) and made a small extra example:
```
Just the page

@tableofcontents{html:6,latex:6,docbook:6}

# SECT1_md

### SECT3_md

## SECT2_md

### SECT3_md1

## SECT2_md2

#### SECT4

# FULL_TREE1

## FULL_TREE2

### FULL_TREE3

#### FULL_TREE4

##### FULL_TREE5

###### FULL_TREE6
```
and the resulting image doesn't look bad but when looking in the md_aa.html file I see:
```
      <li class="level2">
        <a href="#autotoc_md2">SECT2_md</a>
        <ul>
          <li class="level3">
          </li>
          <li class="level3">
            <a href="#autotoc_md3">SECT3_md1</a>
          </li>
        </ul>
      </li>
```
note the:
```
          <li class="level3">
          </li>
```
in my opinion this should not be there (and happens on multiple places).

Another, small thing I see is that in the `doxygen.css` only the `level1` ...`level4` are mentioned, probably when extending the possible depths the `level5` and `level6` have not been added, this can easily be corrected.

**Comment by albert-github** [comment]
Regarding the empty levels in the HTML TOC I've just pushed a proposed patch, pull request #11029.
Also added the level5 and level6 css entry.

**Comment by albert-github** [comment]
Code regarding the empty levels in HTML TOC has been integrated in master on GitHub

**Comment by albert-github** [comment]
Remaining the double `li` (one without one with data) as mentioned in https://github.com/doxygen/doxygen/issues/11016#issuecomment-2249730179.

**Comment by doxygen** [comment]
@albert-github Pushed a commit to fix the empty `li` element.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.12.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
